[Fixes #98400866] [Fixes #98401054] [Fixes #98401064] Bug/fix logging

newOnahtaN opened the pull request:
Fixed logging so that papertrail actually picks up on rails logs (previously did not route to STDOUT in production)
Also some cosmetic changes: 

![alt text](http://vts-monosnap.s3.amazonaws.com/Updates_Since_Last_Deployment_on_070215_-_nathan.owenviewthespace.com_-_VTS_Mail_2015-07-06_12-20-40.png)

newOnahtaN committed:
Fixed href in view and date variable

newOnahtaN committed:
Testing for log information

newOnahtaN committed:
test email

newOnahtaN committed:
Version name

newOnahtaN committed:
CSS formatting

newOnahtaN committed:
testing with puts

newOnahtaN committed:
Added logger to production

newOnahtaN committed:
removed puts

newOnahtaN committed:
Formatting output a little better

newOnahtaN committed:
removed comment

kenliu commented:
Can you remove the "requested by" line, it's not really needed

Also change the heading to be "stories released to production on xxx date"

The email subject should be "production release xxx date vXXXX"

On Monday, July 6, 2015, Nathan Owen notifications@github.com wrote:

> Fixed logging so that papertrail actually picks up on rails logs
> (previously did not route to STDOUT in production)
> Also some cosmetic changes:
> 
> [image: alt text]
> 
> ## https://camo.githubusercontent.com/24786bb7b9af8ae6c214c446dee1dfda787e5689/687474703a2f2f7674732d6d6f6e6f736e61702e73332e616d617a6f6e6177732e636f6d2f557064617465735f53696e63655f4c6173745f4465706c6f796d656e745f6f6e5f3037303231355f2d5f6e617468616e2e6f77656e7669657774686573706163652e636f6d5f2d5f5654535f4d61696c5f323031352d30372d30365f31322d32302d34302e706e67
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/viewthespace/pivotal-deployment-emailer/pull/3
> Commit Summary
> - Fixed href in view and date variable
> - Testing for log information
> - test email
> - Version name
> - CSS formatting
> - testing with puts
> - Added logger to production
> - removed puts
> - Formatting output a little better
> - removed comment
> 
> File Changes
> - _M_ app/mailers/delivered_pivotal_stories_mailer.rb
>   https://github.com/viewthespace/pivotal-deployment-emailer/pull/3/files#diff-0
>   (7)
> - _M_ app/models/delivered_stories_report.rb
>   https://github.com/viewthespace/pivotal-deployment-emailer/pull/3/files#diff-1
>   (5)
> - _M_
>   app/views/delivered_pivotal_stories_mailer/delivered_pivotal_stories_mailer.html
>   https://github.com/viewthespace/pivotal-deployment-emailer/pull/3/files#diff-2
>   (11)
> - _M_ config/environments/production.rb
>   https://github.com/viewthespace/pivotal-deployment-emailer/pull/3/files#diff-3
>   (1)
> 
> Patch Links:
> 
>    -
>    https://github.com/viewthespace/pivotal-deployment-emailer/pull/3.patch
>    -
>    https://github.com/viewthespace/pivotal-deployment-emailer/pull/3.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/viewthespace/pivotal-deployment-emailer/pull/3.

## 

Sent from my mobile

newOnahtaN committed:
Changed some cosmetics and made it so that if there are no stories to report, no email is sent

newOnahtaN committed:
Changed title

newOnahtaN committed:
Added message to empty email and removed attachments

newOnahtaN committed:
Added message to empty email and removed attachments

newOnahtaN committed:
Finished testing for nil stories

newOnahtaN commented:
I just implemented the changes that you suggested, this is what the email looks like now:

![alt text](http://vts-monosnap.s3.amazonaws.com/Production_release_070615_v1786_-_nathan.owenviewthespace.com_-_VTS_Mail_2015-07-07_10-23-55.png)

In addition, I've added a feature where if this script doesn't find any delivered stories, (presumably because the gem that marks them delivered does not do so) it does not send a mass email out, and instead sends a single email to a "backup email" to notify them that there were no stories with the last deployment. I did this to cut out noise from empty deployments.

Currently, that backup email is set in the environment variables as ken@vts.com, but you can very well just remove your email, and the script will run just fine, although no one will be notified when an email is not sent out.

newOnahtaN committed:
centered title
